### PR TITLE
backport: bitcoin#24281, 24360, 24347, 24376, 24370, 24305

### PR DESCRIPTION
**Pull request by vijaydasmp**
Backports

### CODE REVIEW

File: src/bitcoin-wallet.cpp [hunk 1]
```diff
@@ -31,6 +31,7 @@ static void SetupWalletToolArgs(ArgsManager& argsman)
     argsman.AddArg("-dumpfile=<file name>", "When used with 'dump', writes out the records to this file. When used with 'createfromdump', loads the records into a new wallet.", ArgsManager::ALLOW_STRING, OptionsCategory::OPTIONS);
     argsman.AddArg("-debug=<category>", "Output debugging information (default: 0).", ArgsManager::ALLOW_ANY, OptionsCategory::DEBUG_TEST);
     argsman.AddArg("-descriptors", "Create descriptors wallet. Only for 'create'", ArgsManager::ALLOW_BOOL, OptionsCategory::OPTIONS);
+    argsman.AddArg("-legacy", "Create legacy wallet. Only for 'create'", ArgsManager::ALLOW_BOOL, OptionsCategory::OPTIONS);
```
Comment (PastaPastaPasta) [inline src/bitcoin-wallet.cpp:34]: 23002: @knst please evaluate if we are ready for descriptor by default. I thought we had more we needed to fix / implement

Comment (knst) [reply src/bitcoin-wallet.cpp:34]: There is still missing mnemonic feature for descriptor wallets, which is a blocker IMO

I'll work on it for 22.1+

Comment (vijaydasmp) [reply src/bitcoin-wallet.cpp]: dropped 23002

Comment (UdjinM6) [reply src/bitcoin-wallet.cpp]: should drop 24281

File: src/bitcoin-wallet.cpp [hunk 2]
```diff
@@ -56,7 +56,7 @@ static bool WalletAppInit(ArgsManager& args, int argc, char* argv[])
             strUsage += "\n"
                     "dash-wallet is an offline tool for creating and interacting with " PACKAGE_NAME " wallet files.\n"
                     "By default dash-wallet will act on wallets in the default mainnet wallet directory in the datadir.\n"
-                    "To change the target wallet, use the -datadir, -wallet and -testnet/-regtest arguments.\n\n"
+                    "To change the target wallet, use the -datadir, -wallet and -regtest/-signet/-testnet arguments.\n\n"
```
Comment (UdjinM6) [inline src/bitcoin-wallet.cpp:59]: 24376: we have no signet

**Review by UdjinM6 — APPROVED** [review]
utACK e738513a2cb4779ef1d47789db8d1d7eec62e420

**Review by PastaPastaPasta — APPROVED** [review]
utACK e738513a2cb4779ef1d47789db8d1d7eec62e420

**Pull request merged**